app/src/main/java/com/example/taskwise_eventmaster/domain/repository/RoomEventRepository.kt

```diff
@@ -0,0 +1,165 @@
+                city = event.city,
+                country = event.country,
+            )
+
+            dao.upsertEvent(eventPe)
+
+            for (thumbnail in event.thumbnails) {
+
+                val thumbnailPe = ThumbnailPe(
+                    thumbnailUrl = thumbnail.thumbnailUrl,
+                    eventId = event.id,
+                )
+
+                daoThumbnail.upsertThumbnail(thumbnailPe)
+            }
+        } catch (e: Exception) {
+            println("Couldn't upsert an event")
+        }
+    }
+
+    override suspend fun deleteEventLocal(eventId: Int) {
+        try {
+            daoThumbnail.deleteThumbnailsFromEvent(eventId)
+            dao.deleteEvent(eventId)
+        } catch (e: Exception) {
+            println("Error during deleting an event")
+        }
+    }
+
+    override suspend fun getEventRemote(eventId: Int): Event? = withContext(Dispatchers.IO) {
+        try {
+            val eventNe = dataSource.getEvents()?.events?.find { event -> event.id == eventId }
+
+            val thumbnails = getThumbnails(eventId)
+
+            eventNe?.let {
+                Event(
+                    id = eventNe.id,
+                    type = eventNe.type,
+                    name = eventNe.venue.name,
+                    dateTime_Utc = eventNe.datetime_utc.toLocalDateTime().toJavaLocalDateTime(),
+                    address = eventNe.venue.address,
+                    city = eventNe.venue.city,
+                    country = eventNe.venue.country,
+                    thumbnails = thumbnails
+                )
+            }
+
+        } catch (e: Exception) {
+            println("Couldn't get an event with this ID, remote")
+            null
+        }
+    }
+
+    override suspend fun getEventLocal(eventId: Int): Event? = withContext(Dispatchers.IO) {
+        val eventPe = dao.getEventById(eventId)
+
+        val thumbnails = getThumbnails(eventId)
+
+        if (eventPe != null) {
+            Event(
+                id = eventPe.id,
+                type = eventPe.type,
+                name = eventPe.name,
+                dateTime_Utc = eventPe.dateTime_Utc,
+                address = eventPe.address,
+                city = eventPe.city,
+                country = eventPe.country,
+                thumbnails = thumbnails
+            )
+        } else {
+            println("Couldn't get an event with this ID, local")
+            null
+        }
+    }
+
+    override suspend fun getAllEventsRemote(): List<Event> = withContext(Dispatchers.IO) {
+        val collectionOfEventsNe = dataSource.getEvents()
+
+        collectionOfEventsNe?.events?.map { event ->
+
+            //val thumbnails = getThumbnails(event.id) // extract from the db
+
+            var thumbnails = mutableListOf<Event.Thumbnail>()
+
+            for (performer in event.performers) {
+                if (performer.image != null) {
+                    thumbnails += Event.Thumbnail(performer.image)
```

Comment:
You can also use `thumbnails.add(...)`. That way you don't need to make `thumbnails` a `var`